pkg/relayer/miner/miner.go

```diff
@@ -0,0 +1,408 @@
+package miner
+
+import (
+	"context"
+	"crypto/sha256"
+	"hash"
+	"log"
+
+	"cosmossdk.io/depinject"
+
+	"github.com/pokt-network/poktroll/pkg/client"
+	"github.com/pokt-network/poktroll/pkg/either"
+	"github.com/pokt-network/poktroll/pkg/observable"
+	"github.com/pokt-network/poktroll/pkg/observable/channel"
+	"github.com/pokt-network/poktroll/pkg/observable/filter"
+	"github.com/pokt-network/poktroll/pkg/observable/logging"
+	"github.com/pokt-network/poktroll/pkg/relayer"
+	"github.com/pokt-network/poktroll/pkg/relayer/protocol"
+	servicetypes "github.com/pokt-network/poktroll/x/service/types"
+)
+
+var (
+	_             relayer.Miner = (*miner)(nil)
+	defaultHasher               = sha256.New()
+	// TODO_THIS_COMMIT: where (on-chain) should this come from?
+	// TODO_TECHDEBT: setting this to 0 to effectively disable mining for now.
```

Comment:
- Use `TODO_BLOCKER` for higher priority techdebt.
- Keeping it at 0 for now makes sense but it'll be a governance param in the future